Independant development environment and rewrite following review

rmoch opened the pull request:
Includes dev env from https://github.com/openfun/xblock-configurable-lti-consumer/pull/8
and modifications following review on https://github.com/openfun/xblock-configurable-lti-consumer/pull/9 

If accepted, https://github.com/openfun/openedx-docker/pull/64 and https://github.com/openfun/edx-platform/pull/48
will have to be rewrited accordingly

rmoch committed:
✨(core) improve configuration and fallback

Use lti_id field instead of description to ease fallback when component
is not installed
Rewrite settings structure to be specific to configurable_lti_consumer
instead of allowing multiple components configuration as designed in the
first place, also default settings are now part of code.
Raise exeption instead of logging
Better code organisation

jmaupetit commented on config/cms/docker_run_development.py:
```diff
@@ -18,3 +18,50 @@
 
 ALLOWED_HOSTS = ["*"]
+
+SESSION_ENGINE = "django.contrib.sessions.backends.cached_db"
```
I'm not sure this is related to the scope of this PR, right?

jmaupetit commented on configurable_lti_consumer/settings.py:
```diff
@@ -0,0 +1,45 @@
+"""Default settings for Configurable LTI Consumer"""
```
This file is usually named `defaults.py` rather than `settings.py`

jmaupetit commented on config/cms/docker_run_development.py:
```diff
@@ -18,3 +18,50 @@
+CONFIGURABLE_LTI_CONSUMER_SETTINGS = config(
+    "CONFIGURABLE_LTI_CONSUMER_SETTINGS", default=DEFAULT_CONFIGURABLE_LTI_CONSUMER_SETTINGS, formatter=json.loads
+)
+from configurable_lti_consumer import filter_configurable_lti_consumer
```
Can you write import statements on top?

jmaupetit commented on configurable_lti_consumer/load_class.py:
```diff
@@ -0,0 +1,23 @@
+
```
Can you add a module docstring to explain a bit the rationale?

jmaupetit reviewed (commented):
Thank you for this!

I have a few remarks to address:

*  ":ok_hand: (review) misc modifications" is not a relevant commit subject, I think we need to avoid using the :ok_hand: type
* I would prefer that you commit the raw edx `component.py` file and your modifications in a separated commit, to ease modifications tracking
* Can you format your python code with `black` to make our coding style homogeneous over openfun's projects?

rmoch committed:
🔧(edx-platform) add file we need to fork

Add file we need to fork and mount it in edx-platform containers
Hawthorn version at revision 6249d4c06a

rmoch committed:
✨(edx-platform) add buttons displaying configured LTI consumers

jmaupetit commented on README.md:
```diff
@@ -78,55 +78,49 @@ $ make down
+            "accept_grades_past_due": False,
+            "ask_to_send_username": True,
+            "ask_to_send_email": True
 
```
Extra empty row to remove?

jmaupetit commented on README.md:
```diff
@@ -135,4 +129,5 @@ except ImportError:
 For now studio integration is made by inserting `utils.configurable_xblocks`
 function call in `edx-platform/cms/djangoapps/contentstore/views/component.py`.
-It adds to preconfigured xblocks "Advanced" button.
+It adds to preconfigured xblocks to "Advanced" button and remove overrided
```
`s/overrided/overridden/`

jmaupetit replied:
`s/remove/removes/`

jmaupetit commented on README.md:
```diff
@@ -135,4 +129,5 @@ except ImportError:
 function call in `edx-platform/cms/djangoapps/contentstore/views/component.py`.
-It adds to preconfigured xblocks "Advanced" button.
+It adds to preconfigured xblocks to "Advanced" button and remove overrided
+`lti_consumer` component if added to advanced modules.
```
[...] it **it has** been added to [...]

jmaupetit commented on config/cms/docker_run_development.py:
```diff
@@ -20,3 +22,46 @@
+            "accept_grades_past_due": False,
+            "ask_to_send_username": True,
+            "ask_to_send_email": True
+
```
Same extra row :wink:

jmaupetit commented on configurable_lti_consumer/defaults.py:
```diff
@@ -0,0 +1,45 @@
+            "accept_grades_past_due": False,
+            "ask_to_send_username": True,
+            "ask_to_send_email": True
+
```
Idem.

jmaupetit reviewed (approved):
I only have minor remarks, else :+1:
